src/ws-dropdown-menu.js

```diff
@@ -0,0 +1,289 @@
+
+  /**
+   * @type {Object}
+   */
+  static defaultProps = {
+    parent: null,
+    items: [],
+    value: null,
+    filterable: false,
+    filter: null,
+    limit: 10,
+    handle: () => {}
+  };
+
+  /**
+   * @type {Object}
+   */
+  static propTypes = {
+    parent: React.PropTypes.object,
+    items: React.PropTypes.array,
+    filterable: React.PropTypes.bool,
+    filter: React.PropTypes.string,
+    limit: React.PropTypes.number
+  };
+
+  /**
+   * @type {Object}
+   */
+  static contextTypes = {
+    multiple: React.PropTypes.bool
+  };
+
+  /**
+   * @param props {Object}
+   * @param context {Object}
+   * @constructor
+   */
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      filter: props.filter,
+      items: props.items,
+      value: props.value,
+      openSubMenu: null
+    };
+  }
+
+  /**
+   * Gets the current height of the menu
+   * @returns {Number}
+   */
+  getHeight() {
+    return this.menuContainer.clientHeight;
+  }
+
+  /**
+   * Send the new height of this menu after update to the parent.
+   * This will be called when updateFilter did set the new state
+   * @void
+   */
+  componentDidUpdate() {
+    this.props.handle('change-size', this.getHeight());
+  }
+
+  /**
+   * Setting the input value as filter
+   * @param event {KeyboardEvent}
+   * @void
+   */
+  updateFilter(event) {
+    event.stopPropagation();
+    this.setState({filter: event.target.value});
+  }
+
+  /**
+   * If there is a filter active it applies it on the available items
+   * @returns {Array<Object>}
+   */
+  getFilteredItems() {
+    let regex = new RegExp(this.state.filter, 'i');
+    return this.state.items.filter(item => {
+      // Don't show items which doesn't match the filter
+      if (this.props.filterable && this.state.filter && !regex.test(item.label)) {
+        return false;
+      }
+      // Don't show items which were stored as value -> don't show them twice
+      return this.context.multiple ? !item.stored : true;
+    });
+  }
+
+  /**
+   * Deselect all items which are not stored as value. Only relevant for multi select dropdown.
+   * When the dropdown will be closed the state will be restored
+   * @void
+   */
+  clearSelections() {
+    if (this.state.items) {
+      this.state.items.forEach(item => {
+        if (item.selected && !item.stored) {
+          item.selected = false;
+        }
+      });
+    }
+  }
+
+  /**
+   * Handles submit action on multi select drop downs
+   * @void
+   */
+  submit(event) {
+    event.stopPropagation();
+    this.state.value = this.state.items.filter(item => {
+      item.stored = item.selected;
+      return item.selected;
+    });
+    // Propagate new value
+    this.props.handle('change', this.state.value);
+    this.setState(this.state);
+  }
+
+  /**
+   * Handles data propagation from child menus
+   * @param type {String} Should be just show-parent
+   * @param data {*}
+   */
+  handlePropagation = (type, data) => {
```

Comment:
I tried to iterate over all functions and to bind them to the scope but it didn't worked out because this function was on the parent class and therefore not included in for in. So i choosed this way instead of multiple .bind(this)